### PR TITLE
dgs: implement barrier for data-loading status tracking

### PR DESCRIPTION
**Pull request by goldenleaves**

### CODE REVIEW

File: dynamic_graph_service/python/coordinator/http_service.py [hunk 1]
```diff
@@ -83,15 +104,29 @@ def do_GET(self):
       self.send_response(200)
       self.end_headers()
       self.wfile.write(bytes(dl_init_json_str, "UTF-8"))
+    elif self.path.startswith("/admin/barrier-status"):
+      self.send_response(200)
+      self.end_headers()
+      param = self.path.split("?")[1]
```
Comment (Seventeen17) [inline dynamic_graph_service/python/coordinator/http_service.py:110]: Better to catching the exception of index out of range

File: dynamic_graph_service/python/coordinator/http_service.py [hunk 2]
```diff
@@ -48,22 +49,42 @@ def do_POST(self):
...
+      self.end_headers()
+      barrier_name = None
+      dl_count = None
+      dl_id = None
+      params = body.decode("utf-8").split("&")
+      for param in params:
+        kv = param.split("=")
+        if kv[0] == "name":
+          barrier_name = kv[1]
+        elif kv[0] == "dl_count":
```
[the remark this answers is not in the document]
Comment (goldenleaves) [reply dynamic_graph_service/python/coordinator/http_service.py]: okay

File: dynamic_graph_service/python/coordinator/http_service.py [hunk 3]
```diff
@@ -48,22 +49,42 @@ def do_POST(self):
...
+      response.write(bytes("Creating Serving Worker Checkpoint {}.\n"
+                           .format("Successfully" if serving_res else "Failed")))
       self.wfile.write(response.getvalue())
+    elif self.path.startswith("/admin/barrier"):
+      self.send_response(200)
+      self.end_headers()
+      barrier_name = None
+      dl_count = None
+      dl_id = None
+      params = body.decode("utf-8").split("&")
```
[the remark this answers is not in the document]
Comment (goldenleaves) [reply dynamic_graph_service/python/coordinator/http_service.py]: done

File: dynamic_graph_service/python/coordinator/http_service.py (hunk 1 above)
Comment (goldenleaves) [reply dynamic_graph_service/python/coordinator/http_service.py]: using urllib.parse

**Review by Seventeen17 — APPROVED** [review]

**Review by LiSu — APPROVED** [review]
LGTM

**Pull request merged**